src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -9161,12 +9162,29 @@ private ExpressionSyntax ParseTerm(Precedence precedence)
                     // misplaced ::
                     // TODO: this should not be a compound name.. (disallow dots)
                     expr = this.ParseQualifiedName(NameOptions.InExpression);
+                    break;
+                case SyntaxKind.EqualsGreaterThanToken:
+                    return this.ParseLambdaExpression();
```

Comment:
>return [](start = 20, length = 6)

Why do we return in this lambda case but not in the `StaticKeyword` case below?

Reply:
that's a bug, albeit not a major one (since this is error recovery code).  i will fix though to be consistent since that makes a lot more sense.